Makefile

```diff
@@ -0,0 +1,36 @@
+CFLAGS=-O3 -march=native -mtune=native -Wall -Wextra -pedantic -Ithird_party
```

Comment:
Fairly certain `-mtune=native` is implied when using `-march=native` but at this point, we're splitting hairs.

https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html